docs/rules/no-constructor-attributes.md

```diff
@@ -1,39 +1,63 @@
 # Disallows interaction with attributes in constructors (no-constructor-attributes)
 
-Constructors of custom elements should never interact with the DOM or
-attributes as the element may not yet be upgraded.
+Constructors of custom elements should never interact with their own light DOM and attributes as the element will not be connected to the DOM when created programmatically (i.e. using `document.createElement()`).
```

Comment:
nit: can we maintain the line length here? (max 80)

Reply:
also really its not because it won't be connected to the DOM, rather that it may not yet be upgraded. it could be in DOM but not yet upgraded and would break assumptions the consumer might have around createElement semantics.

i'd keep the old wording (upgraded vs connected)